language/move-core/types/src/vm_status.rs

```diff
@@ -235,9 +235,26 @@ pub fn convert_prologue_runtime_error(status: VMStatus) -> VMStatus {
             };
             VMStatus::Error(new_major_status)
         }
-        status @ VMStatus::ExecutionFailure { .. }
-        | status @ VMStatus::Error(_)
+        status @ VMStatus::ExecutionFailure { .. } | status @ VMStatus::Executed => status,
+        VMStatus::Error(code) => {
+            debug_assert!(false, "Unexpected prologue/epilogue error: {:?}", code);
+            VMStatus::Error(StatusCode::UNEXPECTED_ERROR_FROM_PROLOGUE)
+        }
+    }
+}
+
+/// Checks particular Move abort codes to specific validation error codes for the epilogue
+/// Any non-abort non-execution code is considered an invariant violation, specifically
+/// `UNEXPECTED_ERROR_FROM_EPILOGUE`
+pub fn convert_epilogue_error(status: VMStatus) -> VMStatus {
+    match status {
+        status @ VMStatus::MoveAbort(_, _)
+        | status @ VMStatus::ExecutionFailure { .. }
         | status @ VMStatus::Executed => status,
+        VMStatus::Error(code) => {
+            debug_assert!(false, "Unexpected prologue/epilogue error: {:?}", code);
```

Comment:
logging critical here but this looks good!

Reply:
Let me do this in a separate PR. There is a bunch of logging I want to add to this file